test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/blocks/EventBasedGatewayBlockBuilder.java

```diff
@@ -78,11 +80,26 @@ public ExecutionPathSegment findRandomExecutionPath(final Random random) {
         new StepPublishMessage(
             getMessageName(branch), CORRELATION_KEY_FIELD, CORRELATION_KEY_VALUE);
     result.appendDirectSuccessor(executionStep);
-    result.append(blockBuilder.findRandomExecutionPath(random));
+    result.append(blockBuilder.findRandomExecutionPath(random, context));
 
     return result;
   }
 
+  @Override
+  public String getElementId() {
```

Comment:
I have moved some of the generic functionality to an abstract class and gotten rid of the default methods in the interface. 

`getElementId` shouldn't return an optional as a block always has to return an element id now that we do the matching on the starting element on an element id. The `BlockSequenceBuilder` should't really need it, but since we need a way to identify the unique block sequence builders an id will still be generated for these.